src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,219 @@
+                    "indexCheckException: Wrong index < 0 || index > size");
+        }
+    }
+
+    public void indexEqualsSizeCheck(int index) {
+        if (index == size) {
+            throw new IndexOutOfBoundsException("Can`t get element when index == size");
+        }
+    }
+
     @Override
     public void add(T value) {
+        linkLast(value);
     }
 
     @Override
     public void add(T value, int index) {
+        indexCheck(index);
+
+        if (index == 0) {
+            linkFirst(value);
+            return;
+        }
+
+        if (index == size) {
+            linkLast(value);
+            return;
+        }
+
+        linkByIndex(value, index);
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T value : list) {
+            add(value);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        indexCheck(index);
+        indexEqualsSizeCheck(index);
+
+        Node<T> x = findNodeByIndex(index);
+        T oldValue = x.value;
+        return oldValue;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        indexCheck(index);
+        indexEqualsSizeCheck(index);
+        Node<T> x = findNodeByIndex(index);
+        T oldValue = x.value;
+        x.value = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        indexCheck(index);
+        indexEqualsSizeCheck(index);
+        Node<T> x = findNodeByIndex(index);
+        T oldValue = x.value;
+
+        if (x.prev == null && x.next == null) {
+            head = null;
+            tail = null;
+            size--;
+            return oldValue;
+        }
+
+        if (x.prev == null) {
+            head = x.next;
+            head.prev = null;
+            x.next = null;
+            size--;
+            return oldValue;
+        }
+
+        if (x.next == null) {
+            tail = x.prev;
+            tail.next = null;
+            x.prev = null;
+            size--;
+            return oldValue;
+        }
+
+        x.prev.next = x.next;
+        x.next.prev = x.prev;
+        x.prev = null;
+        x.next = null;
+        size--;
+        return oldValue;
     }
 
     @Override
     public boolean remove(T object) {
+        Node<T> currentNode = head;
+        for (int i = 0; i < size; i++) {
+            if ((object == null && currentNode.value == null)
+                    || (object != null && object.equals(currentNode.value))) {
+                remove(i);
+                return true;
+            }
+            currentNode = currentNode.next;
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        if (head == null) {
+            return true;
+        } else {
+            return false;
+        }
```

Comment:
The `isEmpty` method can be simplified to `return head == null;` for conciseness.